lib/graphql/batch/executor.rb

```diff
@@ -0,0 +1,38 @@
+module GraphQL::Batch
+  class Executor
+    THREAD_KEY = :"#{name}.batched_queries"
+    private_constant :THREAD_KEY
+
+    def self.current
+      Thread.current[THREAD_KEY] ||= new
+    end
+
+    attr_reader :loaders
+
+    def initialize
+      @loaders = {}
+    end
+
+    def shift
+      @loaders.shift.last
+    end
+
+    def execute(loader)
```

Comment:
make `execute` and `shift` private? or maybe fold some of that together, right now `execute` could even be static which feels a bit weird